refactor(health-service): simplify health status tracking with global…

ashtonjordan opened the pull request:
… router reference

- Replace instance router reference with global router to avoid variable conflicts
- Remove redundant logging statements from health check updates
- Simplify health status retrieval in monitoring app using get_all_service_statuses()
- Implement status code to name conversion for consistent service status display
- Add degraded state detection when some but not all services are serving
- Remove unnecessary last_check_time tracking from health data response
- Clear datasource_url configuration value for security and flexibility
- Improve overall health determination logic to check both serving count and total services

ashtonjordan committed:
refactor(health-service): simplify health status tracking with global router reference

- Replace instance router reference with global router to avoid variable conflicts
- Remove redundant logging statements from health check updates
- Simplify health status retrieval in monitoring app using get_all_service_statuses()
- Implement status code to name conversion for consistent service status display
- Add degraded state detection when some but not all services are serving
- Remove unnecessary last_check_time tracking from health data response
- Clear datasource_url configuration value for security and flexibility
- Improve overall health determination logic to check both serving count and total services

adamweeks reviewed (changes requested):
Good changes but not quite sure why the global router is needed. I'd prefer to keep the router as an instance variable passed to the class, the way `self.router` is used in `wxcc_gateway_server.py`.

ashjorda commented on src/monitoring/app.py:
```diff
@@ -690,32 +690,31 @@ def get_status_data() -> Dict[str, Any]:
+
+            # Determine overall health
+            overall_healthy = serving_count > 0 and serving_count == total_count
+            grpc_status = "SERVING" if overall_healthy else ("NOT_SERVING" if serving_count == 0 else "DEGRADED")
```
Right on!

ashjorda commented on src/core/health_service.py:
```diff
@@ -31,7 +38,7 @@ def __init__(self, router: Optional[VirtualAgentRouter] = None):
             router: VirtualAgentRouter instance for checking connector health
         """
         super().__init__()
-        self.router = router
```
I concur, see my other comment below.

ashjorda commented on src/core/health_service.py:
```diff
@@ -53,33 +60,28 @@ def _update_service_health(self):
-                # Check if router has available agents
-                if self.router:
-                    available_agents = self.router.get_all_available_agents()
+                # Use global router to avoid variable conflicts
```
Good catch! After reverting to self.router and testing, I found the real issue wasn't a variable conflict at all. The bug was in wxcc_gateway_server.py line 752, where I was passing self.logger instead of self.router to the HealthCheckService constructor. This caused the router to be None, making health checks fail.

The fix was simple: pass the correct parameter. The original instance variable approach with self.router works perfectly and is the cleaner, more Pythonic solution. The global variable pattern was unnecessary. *FacePalm*

ashtonjordan committed:
fix(health-service): correct router initialization bug causing SERVICE_UNKNOWN status

Root Cause:
- WxCCGatewayServer was incorrectly passing  instead of
  to HealthCheckService constructor (line 752)
- This caused health checks to fail since logger objects don't have get_all_available_agents()
- Health status remained stuck at SERVICE_UNKNOWN instead of SERVING

Changes:
- Fix wxcc_gateway_server.py to pass  to HealthCheckService
- Revert unnecessary global router pattern back to cleaner instance variable approach
- Remove  variable and  function from health_service.py
- Restore  instance variable pattern in HealthCheckService
- Update  to use  directly

The global router pattern was introduced while debugging connector count issues,
but it was masking the real bug: passing the wrong argument to the constructor.
The original instance variable approach was correct all along.

Fixes: Health service now correctly reports SERVING status when agents are available

adamweeks reviewed (approved):
LGTM
